chia/wallet/wallet_node.py

```diff
@@ -1289,11 +1289,13 @@ async def fetch_and_validate_the_weight_proof(
 
     async def get_puzzle_hashes_to_subscribe(self) -> List[bytes32]:
         all_puzzle_hashes = list(await self.wallet_state_manager.puzzle_store.get_all_puzzle_hashes())
+        puzzle_store_count = len(all_puzzle_hashes)
         # Get all phs from interested store
         interested_puzzle_hashes = [
             t[0] for t in await self.wallet_state_manager.interested_store.get_interested_puzzle_hashes()
         ]
         all_puzzle_hashes.extend(interested_puzzle_hashes)
+        assert puzzle_store_count + len(interested_puzzle_hashes) == len(set(all_puzzle_hashes))
```

Comment:
as far as I can tell, this assert is trivially true.
you check that extending a list causes its length to be increased by the list you're extending it by.

Did you mean for this to be a set, to ensure there are no duplicates?

Reply:
Did you miss the `set(...)` in `len(set(all_puzzle_hashes))`?

Reply:
oh, I did :)